### PR TITLE
Js us23 24

### PR DESCRIPTION
**Pull request by jrsewell400**
Made the following changes:
- created tests for Cart model and Cart features
- updated seed file
- updated Cart controller with new methods: increase_quanttity, decrease_quantity, to handle increasing and decreasing items in cart
- added new links to Cart Show page to increase and decrease item quantity
- updated routes to include increase and decrease routes to appropriate controller

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -32,4 +32,20 @@ def total
...
+    contents[item_id] -= 1
+  end
+
+  def out_of_stock?(item_id) 
+    Item.find(item_id).inventory <= contents[item_id]
+  end
+
+  def decrease_removal?(item_id)
+    contents[item_id] == 0
+  end
```
Comment (elguapogordo) [inline app/models/cart.rb:49]: `< 1` and `== 0` are only both truthful at one point, remember that :-)

**Review by elguapogordo — APPROVED** [review]
looks great on the server

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -20,8 +20,27 @@ def remove_item
     session[:cart].delete(params[:item_id])
     redirect_to '/cart'
   end
-  private
-  def restrict_admin
-    render file: "/public/404" if current_admin?
+
+  def increase_quantity
+    if cart.out_of_stock?(params[:item_id])
```
Comment (zachholcomb) [inline app/controllers/cart_controller.rb:25]: looks good here like the use of all these helper methods

File: app/models/order.rb [hunk 3]
```diff
@@ -5,6 +5,8 @@ class Order <ApplicationRecord
   has_many :items, through: :item_orders
   belongs_to :user
 
+  enum status: %w( Pending Packaged Shipped Cancelled )
+
```
Comment (zachholcomb) [inline app/models/order.rb:9]: glad we changed this earlier than later

Comment (steve-meyers) [inline app/models/order.rb:9]: Interesting, I wasn't even thinking we could use enums for order status!

File: db/seeds.rb [hunk 4]
```diff
@@ -16,18 +16,18 @@
...
+dog_bowl = dog_shop.items.create(name: "Dog Bowl", description: "Great dog bowl!", price: 7, image: "https://www.talltailsdog.com/pub/media/catalog/product/cache/a0f79b354624f8eb0e90cc12a21406d2/u/n/untitled-6.jpg", inventory: 32)
+dog_ball = dog_shop.items.create(name: "Dog Ball", description: "Awesome dog ball!", price: 5, image: "https://img.chewy.com/is/image/catalog/59155_MAIN._AC_SL1500_V1518033665_.jpg", inventory: 20)
+dog_leash = dog_shop.items.create(name: "Dog Leash", description: "Sturdy dog leash!", price: 10, image: "https://cdn.shopify.com/s/files/1/1728/3089/products/max_and_neo_small_dog_leash_black.jpg?v=1555617800", inventory: 11)
 
 #orders
-order1 = Order.create(name: 'Steve', address: '555 Free St.', city: 'Plano', state: 'TX', zip: '88992')     
-order2 = Order.create(name: 'Steve', address: '555 Free St.', city: 'Plano', state: 'TX', zip: '88992')
+order1 = Order.create(name: 'Steve Meyers', address: '555 Free St.', city: 'Plano', state: 'TX', zip: '88992')     
+order2 = Order.create(name: 'Jordan Sewell', address: '123 Fake St.', city: 'Arvada', state: 'CO', zip: '80301')
```
Comment (zachholcomb) [inline db/seeds.rb:30]: we will need to add users for these orders, orders are now dependent on users to be create

**Review by steve-meyers — APPROVED** [review]
Looks good! I also like the use of all the helper methods to make the cart functionality work!

**Review by zachholcomb — APPROVED** [review]

**Pull request merged**